### PR TITLE
System Test: Module (WAZUH-DB) - The agent with state different than never_connected is added to a "group_test" group by API

### PR DESCRIPTION
**Pull request by CamiRomero**
In this PR the test suite for `The agent with state different than never_connected is added to a "group_test" group by API`  is added. A total of  new test cases covering different modes and combinations of options.

<table>
  <tr>
    <th>Related issues</th>
    <th><a href="[url](https://github.com/wazuh/wazuh-qa/issues/2507)">#2508</a></th>
    <th><a href="[url](https://github.com/wazuh/wazuh/issues/10771)">#10771</a></th>
  </tr>
</table>

## Cases:

- [x] Assign agent with API:
        - The agent is active and registers with group  to the `master` node using API
        - The agent is disconnected and registers with group  to the `master` node using API
       - The agent is active and registers with group  to the `worker` node using API
       - The agent is disconnected and registers with group  to the `worker` node using API

--- 

## Configuration options
In order to run the test, first the environment located in `/test/system/provisioning/enrollment_cluster` must be enabled with:
`sudo ansible-playbook -i inventory.yml playbook.yml --extra-vars='{"wazuh_branch": "dev-10771-agent-groups-files-to-wazuh-db"}'`

**Local Internal Options:** No Local Internal Options used

---

---

## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
| test/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py  | CentOS - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8216742/R1.zip)[:large_blue_circle:][:large_blue_circle:]| 09/03/2022 |  @CamiRomero  | 

---
## Required
- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

**Review by Deblintrake09 — CHANGES_REQUESTED** [review]

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        host_manager.control_service(host='wazuh-agent1', service='wazuh', state="stopped")
+        time.sleep(10)
+        check_agent_status(agent_id, agent_name, agent_ip, 'disconnected', host_manager, test_infra_managers)
+
+    token = host_manager.get_api_token('wazuh-master')
+
+    response = host_manager.make_api_call('wazuh-master', method='PUT',
+                                          endpoint=f'/agents/{agent_id}/group/group_test?pretty=true',
+                                          token=token)
+    print(response)
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py:123]:
```suggestion
    response = host_manager.make_api_call('wazuh-master', method='PUT',
                                          endpoint=f'/agents/{agent_id}/group/group_test?pretty=true',
                                          token=token)
```

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        - The agent 'Agent_name' with ID 'Agent_id' belongs to groups: group_test."
+    '''
+
+    agent_ip, agent_id, agent_name, manager_ip = register_agent(test_infra_agents[0], agent_target, host_manager)
+
+    # Check that agent has no group assigned
+    check_agent_groups(agent_id, 'Null', test_infra_managers, host_manager)
+
+    # Check that agent has client key file
+    assert check_keys_file(test_infra_agents[0], host_manager)
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py:104]: It doesn't have an error message when the assert fails. Is it for some reason?

**Review by damarisg — CHANGES_REQUESTED** [review]

**Review by damarisg — CHANGES_REQUESTED** [review]

**Review by Deblintrake09 — APPROVED** [review]

**Comment by Deblintrake09** [comment]
## Tests Results

| Test Path | Os/Type | Ex
[R1.zip](https://github.com/wazuh/wazuh-qa/files/8256065/R1.zip)
[R2.zip](https://github.com/wazuh/wazuh-qa/files/8256067/R2.zip)
[R3.zip](https://github.com/wazuh/wazuh-qa/files/8256068/R3.zip)
ecution Type | Results | Date | By |
|--|--|--|--|--|--|
| test/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py  | CentOS - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8256065/R1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8256067/R2.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8256068/R3.zip)| 15/03/2022 |  @CamiRomero  |

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (damarisg) [reply tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py]: @Deblintrake09

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    parameters:
+        - agent_target:
+            type: string
+            brief: name of the host where the agent will register
+        - initial_status:
+            type: string
+            brief: status of the agent when the assign the new group
+        - clean_enviroment:
+            type: fixture
+            brief: Reset the wazuh log files at the start of the test. Remove all registered agents from master.
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py:90]: There aren't the variables(test_infra_managers, test_infra_agents, host_manager) defined as parameters of the documentation.

**Review by damarisg — CHANGES_REQUESTED** [review]

**Review by damarisg — APPROVED** [review]

**Pull request merged**